qiskit/quantum_info/synthesis/qsd.py

```diff
@@ -52,11 +60,15 @@ def qs_decomposition(mat, opt_a1=True, decomposer_1q=None, decomposer_2q=None):
        opt_a1 (bool): whether to try optimization A.1 from Shende. This should eliminate 1 cnot
           per call. If True CZ gates are left in the output. If desired these can be further decomposed
           to CX.
+       opt_a2 (bool): whether to try optimization A.2 from Shende. This decomposes two qubit
+          unitaries into a diagonal gate and a two cx unitary and reduces overal cx count by
+          4^(n-2) - 1.
        decomposer_1q (None or Object): optional 1Q decomposer. If None, uses
           :class:`~qiskit.quantum_info.synthesis.one_qubit_decomposer.OneQubitEulerDecomser`
        decomposer_2q (None or Object): optional 2Q decomposer. If None, uses
           :class:`~qiskit.quantum_info.synthesis.two_qubit_decomposer.TwoQubitBasisDecomposer`
           with CXGate.
+       _depth (int): Internal use parameter to track recursion depth.
```

Comment:
I guess pylint doesn't like documenting private keyword arguments. We can just remove this. I know you added it per my suggestion. Personally I thought it'd be better to explain what it was and that users shouldn't touch it, but we can either disable the pylint rule or just remove the arg entry. I'm fine either way.

Reply:
I can remove the arg entry and leave it as a comment